server/service/src/sync/test/integration/central_server_configurations.rs

```diff
@@ -0,0 +1,242 @@
+use std::env;
+
+use rand::{thread_rng, Rng};
+use reqwest::{Client, Url};
+use serde_json::json;
+use util::{hash::sha256, uuid::uuid};
+
+use crate::sync::{settings::SyncSettings, SyncApiV5, SyncConnectionError, SyncCredentials};
+
+impl SyncApiV5 {
+    pub(crate) async fn upsert_central_records(
+        &self,
+        value: &serde_json::Value,
+    ) -> Result<(), SyncConnectionError> {
+        self.create_post("/sync/v5/test/upsert", value)?
+            .send()
+            .await?
+            .error_for_status()?;
+
+        Ok(())
+    }
+}
+
+pub(crate) struct ConfigureCentralServer {
+    api: SyncApiV5,
+    new_site_password: String,
+    server_url: String,
+}
+
+pub(crate) struct CreateSyncSiteResult {
+    pub(crate) new_site_properties: NewSiteProperties,
+    pub(crate) sync_settings: SyncSettings,
+}
+
+impl ConfigureCentralServer {
+    // SYNC_SITE_PASSWORD="pass" SYNC_SITE_NAME="demo" SYNC_URL="http://localhost:2048" NEW_SITE_PASSWORD="pass" cargo test sync_integration_test --features integration_test
+    // OR in VSCODE settings if using rust analyzer:
+    // "rust-analyzer.runnableEnv": { "SYNC_URL": "http://localhost:2048", "SYNC_SITE_NAME": "demo","SYNC_SITE_PASSWORD": "pass", "NEW_SITE_PASSWORD": "pass"}
+    // "rust-analyzer.cargo.features": ["integration_test"]
+    pub(crate) fn from_env() -> ConfigureCentralServer {
+        let password =
+            env::var("SYNC_SITE_PASSWORD").expect("SYNC_SITE_PASSWORD env variable missing");
+        let new_site_password =
+            env::var("NEW_SITE_PASSWORD").expect("NEW_SITE_PASSWORD env variable missing");
+        let site_name = env::var("SYNC_SITE_NAME").expect("SYNC_SITE_NAME env variable missing");
+        let url = env::var("SYNC_URL").expect("SYNC_URL env variable missing");
+
+        ConfigureCentralServer {
+            api: SyncApiV5::new(
+                Url::parse(&url).unwrap(),
+                SyncCredentials {
+                    username: site_name,
+                    password_sha256: sha256(&password),
+                },
+                Client::new(),
+                "",
+            ),
+            server_url: url,
+            new_site_password,
+        }
+    }
+
+    pub(crate) async fn create_sync_site_with_extra_data<F>(
+        &self,
+        // Before site is inserted and linked to store
+        pre_site_creation_data: F,
+    ) -> anyhow::Result<CreateSyncSiteResult>
+    where
+        F: Fn(&NewSiteProperties) -> serde_json::Value,
+    {
+        let new_site_properties = NewSiteProperties::new(&self.new_site_password);
+        self.api
+            .upsert_central_records(&new_site_properties.preliminary_data())
```

Comment:
I think i overcooked this one, will probably change, it should be ok to inset data after site is inserted (was thinking ahead, but too much ahead)